application/forms/Config/UserGroup/AddMemberForm.php

```diff
@@ -133,48 +56,49 @@ public function createElements(array $formData)
      */
     public function onSuccess()
     {
-        $userNames = $this->getValue('user_name') ?: array();
-        if (($users = $this->getValue('users'))) {
-            $userNames = array_merge($userNames, array_map('trim', explode(',', $users)));
-        }
-
-        if (empty($userNames)) {
-            $this->info($this->translate(
-                'Please provide at least one username, either by choosing one '
-                . 'in the list or by manually typing one in the text box below'
-            ));
+        $q = $this->getValue($this->getSearchParameter());
+        if (empty($q)) {
+            Notification::error(t('Please provide at least one username'));
             return false;
         }
 
+        $userNames = array_filter(
```

Comment:
Why are you filtering first and then perform `trim` on each entry again? Isn't it better to perform this the other way round?